### PR TITLE
Added Netty modules

### LINKED ISSUE #331: Support to Spring Webflux (reactive handlers)
Logbook already has Spring support to traditional controllers model. In Spring Boot 2.X was introduced the reactive programming model with spring-webflux. Would be nice have Logbook support to this approach

See more details of Webflux here: https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html#spring-webflux

## Context
I use Logbook for many micro-services in my job. We need to integrate Logbook in some reactive webflux services.

## Possible Implementation
By  WebHandler API? -> https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html#webflux-web-handler-api

Thanks for your attention.
Greetings!

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Motivation and Context

Fixes #331

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    /** Buffer that stores the response content */
+    private final StringBuilder buf = new StringBuilder();
+
+    @Override
+    public void channelReadComplete(final ChannelHandlerContext ctx) {
+        ctx.flush();
+    }
+
+    @Override
+    protected void channelRead0(final ChannelHandlerContext ctx, final Object msg) {
```
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java]: Codacy Issue found: [The method 'channelRead0(ChannelHandlerContext, Object)' has an NPath complexity of 475](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+import io.netty.util.CharsetUtil;
+import org.apiguardian.api.API;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+import static io.netty.handler.codec.http.HttpVersion.*;
```
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java]: Codacy Issue found: [Avoid unused imports such as 'io.netty.handler.codec.http.HttpVersion'](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import io.netty.handler.codec.http.cookie.ServerCookieEncoder;
+import io.netty.util.CharsetUtil;
+import org.apiguardian.api.API;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
```
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java]: Codacy Issue found: [Avoid unused imports such as 'io.netty.handler.codec.http.HttpResponseStatus'](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/main/java/org/zalando/logbook/netty/Conditionals.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package org.zalando.logbook.netty;
+
+import org.zalando.fauxpas.ThrowingConsumer;
+
+final class Conditionals {
```
Comment (whiskeysierra) [inline logbook-netty/src/main/java/org/zalando/logbook/netty/Conditionals.java]: Codacy Issue found: [The utility class name 'Conditionals' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/SequenceTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Test
+    void queuesSecond() {
+        unit.set(1, second);
+
+        verify(second, never()).run();
+    }
+
+    @Test
+    void runsRestEagerly() {
```
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/SequenceTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/IgnoringTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+final class IgnoringTest {
+
+    final AtomicReference<State> state =
+            new AtomicReference<>(new Unbuffered());
+
+    final HttpRequest request = new DefaultHttpRequest(HTTP_1_1, GET, "/");
+
+    @Test
+    void shouldKeepBufferingWhenIgnoring() {
```
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/IgnoringTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/SequenceTest.java (hunk 5 above)
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/SequenceTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java (hunk 3 above)
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java]: Codacy Issue found: [Avoid unused imports such as 'io.netty.handler.codec.http.HttpResponseStatus'](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/main/java/org/zalando/logbook/netty/Conditionals.java (hunk 4 above)
Comment (whiskeysierra) [inline logbook-netty/src/main/java/org/zalando/logbook/netty/Conditionals.java]: Codacy Issue found: [The utility class name 'Conditionals' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java (hunk 2 above)
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java]: Codacy Issue found: [Avoid unused imports such as 'io.netty.handler.codec.http.HttpVersion'](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/IgnoringTest.java (hunk 6 above)
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/IgnoringTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

File: logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java (hunk 1 above)
Comment (whiskeysierra) [inline logbook-netty/src/test/java/org/zalando/logbook/netty/TestHandler.java]: Codacy Issue found: [The method 'channelRead0(ChannelHandlerContext, Object)' has an NPath complexity of 475](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=4982439)

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**